### PR TITLE
Add an enum indicating the timer_mode

### LINKED ISSUE #3151: loss detection timer description could be simplified by definining a timer mode
In `SetLossDetectionTimer` we're setting the loss detection timer in order to:

- trigger an early retransmit (if we received gaps in the acknowledged packets) **or**
- trigger a PTO (if we only have a tail outstanding)

When the timer fires, in `OnLossDetectionTimeout`, we first need to figure out why we set the timer, and trigger an early retransmit or a PTO based on that.

I think the code would be easier if we save a "timer mode" `enum { EARLY_RETRANSMIT, PTO }` in `SetLossDetectionTimer`, and take action based on that mode in `OnLossDetectionTimeout`.

@ianswett, @janaiyengar What do you think?

### PR DESCRIPTION
**Pull request by ianswett**
After reading this, it feels a bit duplicative to whether a loss_time is present to me, but I'm writing a PR so people can take a look.

Fixes #3151

### CODE REVIEW

File: draft-ietf-quic-recovery.md [hunk 1]
```diff
@@ -1151,12 +1157,13 @@ Pseudocode for OnLossDetectionTimeout follows:
...
 OnLossDetectionTimeout():
-  loss_time, pn_space = GetEarliestLossTime()
-  if (loss_time != 0):
-    // Time threshold loss Detection
-    DetectLostPackets(pn_space)
-    SetLossDetectionTimer()
-    return
+  if (timer_mode == LOSS_DETECTION):
+    loss_time, pn_space = GetEarliestLossTime()
+    if (loss_time != 0):
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]: So, this should not be required. Specifically, I think what we might want here is an ASSERT(loss_time != 0), though an assert might be too harsh. Maybe a warning and return.

Also below, under the else for a PTO, add an assert that mode is PTO.

File: draft-ietf-quic-recovery.md [hunk 2]
```diff
@@ -1151,12 +1157,13 @@ Pseudocode for OnLossDetectionTimeout follows:
 
 ~~~
 OnLossDetectionTimeout():
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]: Perhaps rename this to OnTimeout

File: draft-ietf-quic-recovery.md [hunk 3]
```diff
@@ -932,6 +932,10 @@ max_ack_delay:
 loss_detection_timer:
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]: rename this to "timer", with description "multi-modal timer used for loss detection and for sending probes."

**Comment by marten-seemann** [comment]
@ianswett Thanks for the PR! I was planning to submit a PR myself to fix the issue I raised, but I got sidetracked by other stuff.

Don't we also need a timer mode for the amplification limit on the client side?

**Comment by ianswett** [comment]
I am treating(and think of) the client amplification mode as a case of PTO, but it sounds like you're writing a PR, so let's wait for that.

File: draft-ietf-quic-recovery.md (hunk 1 above)
Comment (ianswett) [reply draft-ietf-quic-recovery.md]: I think we either need an if check or an ASSERT.  The following code does not work correctly if you did enter it code and loss_time was 0.

**Pull request closed without merging**